modules/lib/src/main/resources/assets/js/v6/features/shared/dialogs/new-content/NewContentDialogUpload.ts

```diff
@@ -0,0 +1,109 @@
+import {ContentSummaryAndCompareStatus} from 'src/main/resources/assets/js/app/content/ContentSummaryAndCompareStatus';
+import {
+    uploadDataUrlImage,
+    UploadMediaError,
+    uploadMediaFile,
+    UploadMediaSuccess,
+    uploadRemoteImage,
+} from '../../../api/uploadMedia';
+import {
+    addContentTreeUploadItem,
+    removeContentTreeUploadItem,
+    updateContentTreeUploadItemProgress,
+} from '../../../store/contentTreeUpload.store';
+import {generateUniqueName} from '../../../utils/image/generateUniqueName';
+import {reload} from '../../../store/contentTreeLoadingStore';
+import {NotifyManager} from '@enonic/lib-admin-ui/notify/NotifyManager';
+
+type UploadOptions = {
+    dataTransfer: DataTransfer;
+    parentContent?: ContentSummaryAndCompareStatus;
+};
+
+export function uploadMediaFiles({dataTransfer, parentContent}: UploadOptions) {
+    if (dataTransfer.files.length === 0) return;
+
+    const files = Array.from(dataTransfer.files);
+
+    const tasks = files.map((file) => {
+        const id = file.name;
+        const name = file.name;
+        const parentId = parentContent?.getContentId()?.toString();
+
+        addContentTreeUploadItem(id, name, parentId);
+
+        return uploadMediaFile({
+            id,
+            file,
+            parentContent,
+            onProgress: (id, progress) => updateContentTreeUploadItemProgress(id, progress),
+        });
+    });
+
+    tasks.forEach((task) => {
+        task.match(handlers.onEachSuccess, handlers.onEachError);
+    });
```

Comment:
Either explicitly ignore via `void`, or better do:
```ts
await Promise.all(tasks.map(task => task.match(handlers.onEachSuccess, handlers.onEachError)));
```
Then you'll have to mark the function async and add proper return type. You'll ignore it explicitly in the dialog where it is called: `void uploadMediaFiles();`